### PR TITLE
Add dispatch function for gauge conditions

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

This handles forwarding the correct arguments to the gauge condition.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonic.cpp [hunk 1]
```diff
@@ -240,6 +241,14 @@ void test_derived_class(const Mesh<Dim>& mesh) {
           inverse_spatial_metric, spacetime_metric, pi, phi, time,
           inertial_coords);
 
+  // Used dispatch with defaulted arguments that we don't need for Analytic
+  // gauge.
```
Comment (knelli2) [inline tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonic.cpp]: `don't need for DH`

**Review by knelli2 — CHANGES_REQUESTED** [review]
Squash directly.

**Comment by nilsdeppe** [comment]
Fixed, squashed, and rebased. Thanks for the quick review! :D

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Dispatch.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        phi, time, inertial_coords);
+  } else if (const auto* analytic_gauge =
+                 dynamic_cast<const AnalyticChristoffel*>(&gauge_condition);
+             analytic_gauge != nullptr) {
+    analytic_gauge->gauge_and_spacetime_derivative(
+        gauge_h, d4_gauge_h, mesh, time, inertial_coords, inverse_jacobian);
+  } else {
+    ERROR(
+        "Failed to dispatch to Harmonic, DampedHarmonic, or Analytic gauge "
+        "condition.");
```
Comment (knelli2) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Dispatch.cpp]: [Optional] Add `LCOV` statements.

**Review by knelli2 — DISMISSED** [review]

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Dispatch.cpp (hunk 2 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Dispatch.cpp]: Done!

**Review by knelli2 — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**